modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -3634,19 +3639,118 @@ def groupby(
         if not isinstance(by, list):
             by = [by]
 
+        skip_on_aligning_flag = "__skip_me_on_aligning__"
+
         def apply_func(df):  # pragma: no cover
             if any(is_categorical_dtype(dtype) for dtype in df.dtypes[by].values):
                 raise NotImplementedError(
                     "Reshuffling groupby is not yet supported when grouping on a categorical column. "
                     + "https://github.com/modin-project/modin/issues/5925"
                 )
-            return operator(df.groupby(by, **kwargs))
+            result = operator(df.groupby(by, **kwargs))
+            if (
+                align_result_columns
+                and df.empty
+                and result.empty
+                and df.columns.equals(result.columns)
+            ):
+                # We want to align columns only of those frames that actually performed
+                # some groupby aggregation, if an empty frame was originally passed
+                # (an empty bin on reshuffling was created) then there were no groupby
+                # executed over this partition and so it has incorrect columns
+                # that shouldn't be considered on the aligning phase
+                result.attrs[skip_on_aligning_flag] = True
```

Comment:
Is it possible to avoid usage of [experimental](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.attrs.html?highlight=attr#pandas-dataframe-attrs) `attrs`?

Reply:
I don't know of another way of how we can pass this information with the frame. Everything else that comes to my mind seems much more complex

Reply:
ok for now